include/range/v3/algorithm/ends_with.hpp

```diff
@@ -0,0 +1,92 @@
+/// \file
+// Range v3 library
+//
+//  Copyright Johel Guerrero 2019
+//
+//  Use, modification and distribution is subject to the
+//  Boost Software License, Version 0.0. (See accompanying
+//  file LICENSE_0_0.txt or copy at
+//  http://www.boost.org/LICENSE_0_0.txt)
+//
+// Project home: https://github.com/ericniebler/range-v3
+//
+#ifndef RANGES_V3_ALGORITHM_ENDS_WITH_HPP
+#define RANGES_V3_ALGORITHM_ENDS_WITH_HPP
+
+#include <utility>
+#include <concepts/concepts.hpp>
+#include <range/v3/algorithm/result_types.hpp>
+#include <range/v3/detail/config.hpp>
+#include <range/v3/functional/comparisons.hpp>
+#include <range/v3/functional/identity.hpp>
+#include <range/v3/functional/invoke.hpp>
+#include <range/v3/iterator/concepts.hpp>
+#include <range/v3/iterator/operations.hpp>
+#include <range/v3/range/access.hpp>
+#include <range/v3/range/concepts.hpp>
+#include <range/v3/range/dangling.hpp>
+
+namespace ranges
+{
+    /// \addtogroup group-algorithms
+    /// @{
+    template<typename I0, typename I1>
+    using ends_with_result = detail::bool_in1_in2_result<I0, I1>;
+
+    struct ends_with_fn
+    {
+    private:
+        template<typename I0, typename S0, typename I1, typename S1,
+            typename C, typename P0, typename P1>
+        constexpr /*c++14*/ ends_with_result<I0, I1> nocheck(
+            I0 begin0, S0 end0, I1 begin1, S1, C pred, P0 proj0, P1 proj1) const
+        {
+            for(; begin0 != end0; ++begin0, ++begin1)
```

Comment:
Note that I removed the bounds check for the second range in this code I got from `equal.hpp`. Is that okay? The ranges are supposed to be of the same size.

Reply:
I'd be much more comfortable if we checked both ranges.